### PR TITLE
fix(web, api, ee-billing): Fetch current plan from db

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

Current organization plan (aka apiServiceLevel) is not migrated to Clerk for Novu EE. That is we can't do currentOrganization.apiServiceLevel anymore unless we pass it to the Clerk organization public metadata.

Since current plan need to stay in Mongo as well, we fetch it separately from a dedicated API endpoint.

### CODE REVIEW

File: apps/web/src/ee/billing/components/PlanHeader.tsx [hunk 1]
```diff
@@ -24,21 +25,24 @@ export const PlanHeader = () => {
...
-      setApiServiceLevel(currentOrganization.apiServiceLevel || ApiServiceLevelEnum.FREE);
-    }
-  }, [currentOrganization, isOrganizationLoaded]);
+  /*
+   * useEffect(() => {
+   *   if (isOrganizationLoaded) {
+   *     setApiServiceLevel(subscription?.apiServiceLevel);
+   *   }
+   * }, [currentOrganization, isOrganizationLoaded, subscription?.apiServiceLevel]);
+   */
```
Comment (rifont) [inline apps/web/src/ee/billing/components/PlanHeader.tsx]: Should this be deleted now?

**Review by rifont — APPROVED** [review]

**Comment by rifont** [comment]
I approved, but it appears the worker is failing to start in CI tests - https://github.com/novuhq/novu/actions/runs/10206633881/job/28239968368?pr=6230#step:7:235

We need to fix that before merging.

File: apps/web/public/env-config.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ window._env_ = Object.assign(
     SKIP_PREFLIGHT_CHECK: 'true',
     REACT_APP_ENVIRONMENT: 'dev',
     REACT_APP_VERSION: '$npm_package_version',
-    IS_IMPROVED_ONBOARDING_ENABLED: 'false',
+    IS_V2_ENABLED: 'true',
+    IS_V2_EXPERIENCE_ENABLED: 'true',
```
Comment (denis-kralj-novu) [inline apps/web/public/env-config.js:7]: ❓ **question:** ‏ is this change intentional

File: apps/web/src/ee/billing/components/PlanHeader.tsx (hunk 1 above)
Comment (denis-kralj-novu) [reply apps/web/src/ee/billing/components/PlanHeader.tsx]: removed

File: apps/web/public/env-config.js (hunk 2 above)
Comment (SokratisVidros) [reply apps/web/public/env-config.js:7]: Yes. I cleaned the file and sorted flags alphabetically.

File: apps/web/src/ee/clerk/providers/EnterpriseAuthProvider.tsx [hunk 3]
```diff
@@ -16,6 +16,18 @@ const asyncNoop = async () => {};
...
+   * Provide strong custom types for the Clerk organization.publicMetadata object.
+   */
+  interface OrganizationPublicMetadata {
+    defaultLocale: string;
+    domain: string;
+    productUseCases: ProductUseCases;
+    language: string[];
+  }
+}
+
```
Comment (ChmaraX) [inline apps/web/src/ee/clerk/providers/EnterpriseAuthProvider.tsx]: This is already done here: `src/ee/clerk/utils/environment.ts`

```typescript
import {
  UserPublicMetadata as _UserPublicMetadata,
  OrganizationPublicMetadata as _OrganizationPublicMetadata,
} from '@novu/shared';

declare global {
  interface UserPublicMetadata extends _UserPublicMetadata {}

  interface OrganizationPublicMetadata extends _OrganizationPublicMetadata {}
}
```

**Review by ChmaraX — APPROVED** [review]
